### DISCUSSION TITLE
Access to cgroups

### DISCUSSION THREAD

**Discussion by FFY00** [community/community#26507] [Actions]
<p>Right now there is no access to cgroups in the actions. Some things like <code>systemd-nspawn</code> require it so it would be great to have it. We just need access to the <code>/sys/fs/cgroup</code> of the host, this can be easily done with a docker mount point.</p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/ffy00">@ffy00</a> ,</p>
<p>Does your job run on ubuntu-lastest or your self-hosted runner? Could you please share your workflow run logs here? What’s the error message you got? </p>
<p>I tried changing directory to /sys/fs/cgroup and list content of it .  There is no error. </p>
<pre><code>jobs:
  check_cgroup:
**runs-on: ubuntu-latest** steps:
    - name: check cgroup
      run: |
        cd /sys/fs/cgroup
        ls
</code></pre>
<p><img src="https://user-images.githubusercontent.com/24806478/181098095-dd6a1e1c-fae9-4d47-80a6-5dc94b16048f.png" alt="cgroup.png" data-base62-sha1="acr6QvY84DwfWkxDSVVwqWc3k6d" width="299" height="427"></p>

**Comment by FFY00** [comment]
<p>Right, it exists but it does not provide acess to the host cgroup. In my case I need /sys/fs/cgroup/systemd/docker.</p>
<p>The solution would be to pass ‘-v /sys/fs/cgroup:/sys/fs/cgroup’ but ‘-v /sys/fs/cgroup/systemd/docker:/sys/fs/cgroup/systemd/docker’ is enough in my situation.</p>
<p>Example Github Actions run: <a href="https://github.com/FFY00/build-arch-package/runs/589737885" rel="nofollow noopener">https://github.com/FFY00/build-arch-package/runs/589737885</a><br>
Example similar successful Travis run: <a href="https://travis-ci.com/github/FFY00/prjtrellis/builds/160309099" rel="nofollow noopener">https://travis-ci.com/github/FFY00/prjtrellis/builds/160309099</a></p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/ffy00">@ffy00</a> ,</p>
<p>Thank you for your workflow run logs. </p>
<p>According to the GitHub Actions workflow run logs, </p>
<pre><code>(12/12) Rebuilding certificate stores...
Failed to chown() cgroup /sys/fs/cgroup/systemd/actions_job/b8aff34842dca0091d646cfa4bb98189852f3f42f5fb0a49e5a34b8541f91bce/actions_job/b8aff34842dca0091d646cfa4bb98189852f3f42f5fb0a49e5a34b8541f91bce/payload: No such file or directory
Attempted to remove disk file system under "/run/systemd/nspawn/propagate/root", and we can't allow that.
</code></pre>
<p>the file need to be access is under “/sys/fs/cgroup/systemd/actions_job” not “/sys/fs/cgroup/systemd/docker”. This is different between GitHub Actions and Travis.</p>
<p>In GitHub Action workflow , you could try </p>
<pre><code>-v /sys/fs/cgroup/systemd/actions_job:/sys/fs/cgroup/systemd/actions_job
</code></pre>

**Comment by FFY00** [comment]
<p>Right, of course. I don’t know how I missed that. Sorry for bothering.</p>
